File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    double l = count_letters(text);
    double w = count_words(text);
    double s = count_sentences(text);

    int index = round(0.0588 * ((l / w) * 100.0) - 0.296 * ((s / w) * 100.0) - 15.8);
    if (index >= 1 && index < 16)
    {
        printf("Grade %i\n", index);
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

int count_letters(string text)
{
    int letter_count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (islower(text[i]) || isupper(text[i]))
        {
            letter_count++;
        }
    }
    return letter_count;
}

int count_words(string text)
{
    int word_by1 = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            word_by1++;
        }
    }
    word_by1++;
    return word_by1;
}

int count_sentences(string text)
{
    int sentence_count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentence_count++;
        }
    }
    return sentence_count;
}

File: readability.py
def main():
    text = input("Text: ")
    l = count_letters(text)
    w = count_words(text)
    s = count_sentences(text)

    index = round(0.0588 * ((l / w) * 100) - 0.296 * ((s / w) * 100) - 15.8)

    if 1 <= index < 16:
        print(f"Grade {index}")
    elif index >= 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")


def count_letters(text):
    letterCount = 0
    for letter in text:
        if letter.isalpha() == True:
            letterCount += 1
    return letterCount


def count_words(text):
    word_by_1 = 0
    for letter in text:
        if letter == " ":
            word_by_1 += 1
    word = word_by_1 + 1
    return word


def count_sentences(text):
    sentence = 0
    for letter in text:
        match letter:
            case "." | "?" | "!":
                sentence += 1
    return sentence


if __name__ == "__main__":
    main()
